src/main/java/thePackmaster/actions/boardgamepack/RollDiceAction.java

```diff
@@ -0,0 +1,53 @@
+package thePackmaster.actions.boardgamepack;
+
+import com.megacrit.cardcrawl.actions.AbstractGameAction;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+import static thePackmaster.SpireAnniversary5Mod.modID;
+
+public class RollDiceAction extends AbstractGameAction {
+
+    int advantage;
+    List<Integer> dice = new ArrayList<>();
+
+    public RollDiceAction(List<Integer> dice) {
+        this(dice, 0);
+    }
+
+    public RollDiceAction(List<Integer> dice, int advantage) {
+        for (int diceSize: dice)
+            this.dice.add(diceSize);
+        this.advantage = advantage;
+    }
+
+
+    public void update() {
+        //ADD DICE ROLL VISUALS HERE
+//        int diceTotal = 0;
+        for(int numDice = 0; numDice < dice.size(); numDice++)
+        {
+            int curRoll = 0;
+            for(int roll = 0; roll <= advantage; roll++)
+            {
+                int newRoll = AbstractDungeon.miscRng.random(dice.get(numDice) - 1) + 1;
```

Comment:
Wouldn't cardRandomRng fit better here? It's used for basically all impactful mid-combat randomization